pkg/vault/config.go

```diff
@@ -16,18 +16,31 @@ type Config struct {
 	Type       VaultType
 }
 
+func setupViper() {
+	viper.BindEnv("VaultType", "AVP_TYPE")
```

Comment:
We should be able to use the `SetEnvPrefix` to simplify this.
https://github.com/spf13/viper#working-with-environment-variables

Reply:
I looked at that but:

> If the ENV variable name is not provided, then Viper will automatically assume that the ENV variable matches the following format: prefix + "_" + the key name in ALL CAPS. When you explicitly provide the ENV variable name (the second parameter), it does not automatically add the prefix.

We'd have to either rename the Viper keys from `RoleID` to `Role_ID` to match the env var name `AVP_ROLE_ID` or rename the env variables. We can go either way I just didn't do them.

Reply:
Depending on how we answer this, we may or may not need to add docs that say what the accepted keys in a config file /Secret are

Reply:
> rename the Viper keys from RoleID to Role_ID
I am fine with this